Doc/c-api/frame.rst

```diff
@@ -0,0 +1,75 @@
+.. highlight:: c
+
+Frame Objects
+-------------
+
+.. c:type:: PyFrameObject
+
+   The C structure of the objects used to describe frame objects.
+
+   The structure is only part of the internal C API: fields should not be
+   access directly. Use getter functions like :c:func:`PyFrame_GetCode` and
+   :c:func:`PyFrame_GetBack`.
+
+   Debuggers and profilers can use the internal C API to access this structure
+   without calling functions, but the internal C API doesn't provide any
+   backward compatibility warranty.
+
+   .. versionchanged:: 3.11
+      The structure moved to the internal C API headers.
+
+Public members:
+
+* ``f_back`` (read only): Next outer frame object (this frame's caller).
+  See also: :c:func:`PyFrame_GetBack`.
+* ``f_builtins`` (read only): Builtins namespace seen by this frame.
+* ``f_code`` (read only): Code object being executed in this frame.
+  See also :c:func:`PyFrame_GetCode`.
+* ``f_globals`` (read only): Global namespace seen by this frame.
+* ``f_lasti`` (read only): Index of last attempted instruction in bytecode.
+* ``f_lineno``: Current line number in Python source code.
+  See also :c:func:`PyFrame_GetLineNumber`.
+* ``f_locals`` (read only): Local namespace seen by this frame.
+* ``f_trace_lines``: Emit ``PyTrace_LINE`` trace events?
+* ``f_trace_opcodes``: Emit ``PyTrace_OPCODE`` trace events?
+* ``f_trace``: Tracing function for this frame, or ``None``
+
+The :c:func:`PyObject_GetAttrString` and :c:func:`PyObject_SetAttrString`
```

Comment:
I plan to add getters for the remaining attributes that don't have them, and setters for those values that are mutable.
We already have the code, we just need to publish the API.

We need to provide the lower level API, as Cython (and probably others) won't use `PyObject_GetAttrString` or `PyObject_SetAttrString` as they are perceived to be too slow.

Reply:
:+1: